modules/gateway/peers.go

```diff
@@ -514,7 +514,7 @@ func (g *Gateway) permanentPeerManager(closedChan chan struct{}) {
 		// Also wait a while and try again if there is at least one outbound
 		// peer and the selected peer is local. We do not want all of our
 		// outbound peers to be local peers.
-		if err != nil || (numOutboundPeers > 0 && addr.IsLocal() && build.Release != "testing") {
+		if err != nil || (numOutboundPeers >= maxLocalOutboundPeers && addr.IsLocal() && build.Release != "testing") {
```

Comment:
Can you explain this change? `numOutboundPeers` seems to be the total number of outbound peers, local or not. Why compare this to `maxLocalOutboundPeers`?